spec/controllers/pages_controller_spec.rb

```diff
@@ -0,0 +1,50 @@
+RSpec.describe PagesController do
+  describe '#page_update_params' do
+    let(:page) { service.find_page_by_url('/name') }
+    before do
+      controller.instance_variable_set(:@page, page)
+      allow(controller).to receive(:params).and_return(
+        ActionController::Parameters.new(page: params)
+      )
+      allow(controller).to receive(:service_metadata).and_return(service_metadata)
+    end
+
+    context 'when components are present' do
+      let(:component_params) do
+        {
+          'items' => [
+            { 'value' => 'Option 1' },
+            { 'value' => 'Option 2' }
+          ]
+        }
+      end
+      let(:params) do
+        {
+          components: { '0' => JSON.dump(component_params) }
+        }
+      end
+
+      it 'parses components as json' do
+        expect(controller.page_update_params).to eq({
+          'id' => page.id,
+          'latest_metadata' => service_metadata,
+          'components' => [component_params],
+          'service_id' => service.service_id
+        })
+      end
```

Comment:
is it worth testing with two sets of components at this point or wait until the multiple question page is added?

Reply:
Yes I think it is worth to test. Added now